framework/include/utils/RankTwoTensor.h

```diff
@@ -52,6 +52,7 @@ class RankTwoTensor
     initIdentity
   };
 
+  RankTwoTensor * cosserat = NULL;
```

Comment:
This isn't a comment about the approach but more towards starting to push you to think in C++11 terms. Depending on your use you might choose one of the new smart pointers instead of using a raw pointer. An internally managed pointer like this is a perfect candidate for `std::unique_ptr<>` which is as fast as a raw pointer but doesn't have the possibility of leaking. 

Even if you do use a raw pointer the preferred method for initializing them is to use `nullptr` instead of `NULL`. There are sound reasons for that which I won't elaborate here.